packages/automerge-repo/src/DocHandle.ts

```diff
@@ -255,118 +174,161 @@ export class DocHandle<T> //
+  ) {
     try {
       // wait for the document to enter one of the desired states
       await this.#statePromise(awaitStates)
     } catch (error) {
-      // if we timed out (or have determined the document is currently unavailable), return undefined
+      // if we timed out, return undefined
       return undefined
     }
     // Return the document
     return !this.isUnavailable() ? this.#doc : undefined
   }
 
   /**
-   * Returns the current state of the Automerge document this handle manages, or undefined.
-   * Useful in a synchronous context. Consider using `await handle.doc()` instead, check `isReady()`,
-   * or use `whenReady()` if you want to make sure loading is complete first.
+   * Synchronously returns the current state of the Automerge document this handle manages, or
+   * undefined. Consider using `await handle.doc()` instead. Check `isReady()`, or use `whenReady()`
+   * if you want to make sure loading is complete first.
+   *
+   * Not to be confused with the SyncState of the document, which describes the state of the
+   * synchronization process.
    *
-   * Do not confuse this with the SyncState of the document, which describes the state of the synchronization process.
+   * Note that `undefined` is not a valid Automerge document, so the return from this function is
+   * unambigous.
    *
-   * Note that `undefined` is not a valid Automerge document so the return from this function is unambigous.
-   * @returns the current document, or undefined if the document is not ready
+   * @returns the current document, or undefined if the document is not ready.
    */
-  docSync(): A.Doc<T> | undefined {
-    if (!this.isReady()) {
-      return undefined
-    }
-
-    return this.#doc
+  docSync() {
+    if (!this.isReady()) return undefined
+    else return this.#doc
   }
 
-  /** `update` is called by the repo when we receive changes from the network
+  /**
+   * Called by the repo when we receive changes from the network.
    * @hidden
-   * */
+   */
   update(callback: (doc: A.Doc<T>) => A.Doc<T>) {
-    this.#machine.send(UPDATE, {
-      payload: { callback },
-    })
+    this.#machine.send({ type: UPDATE, payload: { callback } })
   }
 
-  /** `setRemoteHeads` is called by the repo either when a doc handle changes or we receive new remote heads
+  /**
+   * Called by the repo either when a doc handle changes or we receive new remote heads.
    * @hidden
    */
   setRemoteHeads(storageId: StorageId, heads: A.Heads) {
     this.#remoteHeads[storageId] = heads
     this.emit("remote-heads", { storageId, heads })
   }
 
-  /** Returns the heads of the storageId */
+  /** Returns the heads of the storageId. */
   getRemoteHeads(storageId: StorageId): A.Heads | undefined {
     return this.#remoteHeads[storageId]
   }
 
-  /** `change` is called by the repo when the document is changed locally  */
+  /** Called by the repo when the document is changed locally.  */
```

Comment:
I think this comment is wrong -- this is the public method and one of a small number of the key methods on the object.

Reply:
Added a more fulsome comment.